Mathlib/Analysis/SpecialFunctions/Log/Deriv.lean

```diff
@@ -249,6 +250,92 @@ theorem abs_log_sub_add_sum_range_le {x : ℝ} (h : |x| < 1) (n : ℕ) :
+  refine ((((((hasDerivAt_id _).const_add _).div ((hasDerivAt_id _).const_sub _) (by grind)).log
+          ?_).const_mul _).sub (HasDerivAt.fun_sum fun i hi ↦ (hasDerivAt_pow _ _).div_const _))
+        |>.congr_deriv ?_
+  · simp only [id_eq, div_ne_zero_iff, Pi.div_apply]; grind
+  have : (∑ i ∈ range n, (2*i+1) * y ^ (2*i) / (2*i+1)) = (∑ i ∈ range n, (y^2) ^ i) := by
+    congr with i
+    simp [field, mul_comm, ← pow_mul]
+  have hy₃ : y ^ 2 ≠ 1 := by simp [hy₁.ne', hy₂.ne]
+  have hy₄ : (1 - y) * (1 + y) = 1 - y ^ 2 := by ring
+  simp [this, field, geom_sum_eq hy₃, hy₄, sub_ne_zero_of_ne, hy₃.symm]
+  ring
+
+/-- A lemma estimating the difference between `1/2 * log ((1+x)/(1+x))` and its
+Taylor series at `0`, where the bound tends to `0`. This bound is particularly useful for explicit
+estimates of logarithms.
+
+Note that thanks to the geometric series, the derivative has a particularly simple form, and means
+that it is more convenient to avoid Taylor's theorem for this proof.
+-/
+lemma sum_range_sub_log_div_le {x : ℝ} (h : |x| < 1) (n : ℕ) :
+    |1 / 2 * log ((1 + x) / (1 - x)) - ∑ i ∈ range n, x ^ (2 * i + 1) / (2 * i + 1)| ≤
+      |x| ^ (2 * n + 1) / (1 - x ^ 2) := by
+  let F (x : ℝ) : ℝ :=
+    1 / 2 * log ((1 + x) / (1 - x)) - (∑ i ∈ range n, x ^ (2 * i + 1) / (2 * i + 1))
+  let F' (y : ℝ) : ℝ := (y ^ 2) ^ n / (1 - y ^ 2)
+  have hI : Icc (-|x|) |x| ⊆ Ioo (-1 : ℝ) 1 := Icc_subset_Ioo (by simp [h]) h
+  -- First step: compute the derivative of `F`
+  have A : ∀ y ∈ Ioo (-1 : ℝ) 1, HasDerivAt F (F' y) y := by
+    intro y hy
+    exact sum_range_sub_log_div_le_aux _ (by grind) (by grind)
+  -- second step: show that the derivative of `F` is small
+  have B : ∀ y ∈ Set.Icc (-|x|) |x|, ‖F' y‖ ≤ |x| ^ (2 * n) / (1 - x ^ 2) := fun y hy ↦ by
+    have : y ^ 2 ≤ x ^ 2 := sq_le_sq.2 (abs_le.2 hy)
+    calc
+    ‖F' y‖ = (y ^ 2) ^ n / |1 - y ^ 2| := by simp [F']
+    _ = (y ^ 2) ^ n / (1 - y ^ 2) := by rw [abs_of_pos (by simpa [abs_lt] using hI hy)]
+    _ ≤ (x ^ 2) ^ n / (1 - x ^ 2) := by gcongr ?_ ^ n / (1 - ?_); simpa [abs_lt] using h
+    _ ≤ |x| ^ (2 * n) / (1 - x ^ 2) := by simp [pow_mul]
+  -- third step: apply the mean value inequality
+  have C : ‖F x - F 0‖ ≤ |x| ^ (2 * n) / (1 - x^2) * ‖x - 0‖ :=
+    (convex_Icc (-|x|) |x|).norm_image_sub_le_of_norm_hasDerivWithin_le
+      (fun y hy ↦ (A _ (hI hy)).hasDerivWithinAt) B
+      (by simp) (by simp [le_abs_self, neg_le, neg_le_abs x])
+  -- fourth step: conclude by massaging the inequality of the third step
+  simpa [F, pow_succ, div_mul_eq_mul_div] using C
+
+/--
+For `0 ≤ x < 1`, the partial sums of the series expansion of `1/2 * log ((1+x)/(1-x))` at `0`
+form a lower bound for it. This shows that the absolute value in `sum_range_sub_log_div_le` can be
+dropped, and gives explicit lower bounds for logarithms.
+-/
+lemma sum_range_le_log_div {x : ℝ} (h₀ : 0 ≤ x) (h : x < 1) (n : ℕ) :
+    ∑ i ∈ range n, x ^ (2 * i + 1) / (2 * i + 1) ≤ 1 / 2 * log ((1 + x) / (1 - x)) := by
```

Comment:
First, do you think it would be better to phrase the statement as `2 * ∑ ... ≤ log ...`? This is only a minor point.

Second, given that the purpose to provide explicit bounds, do you think it makes sense to add one more lemma of the form `1 / 2 * log ... ≤ ∑ ... + ...`? Otherwise if someone wants to use this, they have to untangle it from these two results themselves.

Reply:
I've added the second lemma you mentioned. For the first, I'd prefer to keep it as-is, since then it's exactly the artanh function, rather than a multiple of it.

Reply:
That makes sense.